### PR TITLE
Update to use drone-plugin-lib

### PR DESCRIPTION
**Pull request by donny-dont**
This updates the `drone-manifest` plugin to use the `drone-plugin-lib` and the `boilr-plugin` for scaffolding.

### CODE REVIEW

File: plugin/impl.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+// Execute provides the implementation of the plugin.
+func (p *Plugin) Execute() error {
+	// Anonymous struct for the templating engine
+	var t struct {
+		Build struct {
+			Tag  string
+			Tags []string
+		}
+	}
```
Comment (donny-dont) [inline plugin/impl.go:72]: The thing I'm most concerned about here is the template strings. I'm assuming that everyone just uses `build.tag` and `build.tags` like our manifest files.

**Comment by donny-dont** [comment]
@tboerger would be great to land this 😄

File: plugin/impl.go (hunk 1 above)
Comment (tboerger) [reply plugin/impl.go:72]: tag and tags is important, but i would prefer to provide all available variables within the template.

**Review by tboerger — APPROVED** [review]
Just a small comment, beside that LGTM

**Comment by donny-dont** [comment]
Closing in favor of #40

**Pull request closed without merging**